### PR TITLE
Scan all JSON files (close #779)

### LINKED ISSUE #779: Scan all JSON files, not just manifest.json
**Steps to reproduce:**
Submit the attached webextension:
[iCloud-Bookmarks_v1.4.20.zip](https://github.com/mozilla/addons-linter/files/331481/iCloud-Bookmarks_v1.4.20.zip)

**Expected results:**
A validation error appears explaining that a json file contains block comments and they  should be removed or replaced with “//”.

**Actual results:**
The webextension passes the validation without any error, but it cannot be installed because appears to be corrupted.

**Additional notes:**
- This issue reproduces on AMO-dev, AMO-stage and AMO-production using Firefox 50.0a1 (2016-06-23) under Windows 10 64-bit.
- After submitting and installing the webextension, several errors appear in Browser Console such as: 1466754498299	addons.webextension.<unknown>	ERROR	Loading extension 'null': Loading locale file _locales/nl/messages.json: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data     Log.jsm:753

### PR DESCRIPTION
**Pull request by tofumatt**
This turned out to be a lot more work than I thought it'd be, because of how we were scanning manifest.json files.

Now we scan all JSON files, so anything that would throw Firefox for a loop should be caught by the linter.

### CODE REVIEW

File: src/parsers/manifestjson.js [hunk 1]
```diff
@@ -185,14 +125,12 @@ export default class ManifestJSONParser {
...
       this.collector.addError(messages.MANIFEST_UPDATE_URL);
-      isValid = false;
+      this.isValid = false;
     }
 
     if (isToolkitVersionString(this.parsedJSON.version)) {
       this.collector.addNotice(messages.PROP_VERSION_TOOLKIT_ONLY);
     }
-
-    return isValid;
```
Comment (muffinresearch) [inline src/parsers/manifestjson.js]: Presumably nothing was using this return value?

File: src/parsers/json.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    // Provides ability to directly add messages to
+    // the collector.
+    this.collector = collector;
+
+    // Set the filename for this file
+    this.filename = filename;
+
+    // Any time parsing or validation fails this is set to `false`, so we start
+    // with the assumption it's `true`.
```
Comment (muffinresearch) [inline src/parsers/json.js]: But it's null :)

File: tests/parsers/test.json.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+  it("doesn't evaluate JS code even though esprima is used", () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = [
+      '{',
+      '// Required',
+      '"manifest_version": 2,',
+      '"name": "My Extension",',
+      '// Make the hell sure to use semvar.org if increasing this',
+      '"version": eval("alert(\'uh-oh\')")',
```
Comment (muffinresearch) [inline tests/parsers/test.json.js]: I see this is a moved test and not new, but if the alert was fired, would it fail the test?

File: src/parsers/manifestjson.js [hunk 4]
```diff
@@ -1,103 +1,43 @@
...
 import { isToolkitVersionString } from 'schema/formats';
 import { singleLineString } from 'utils';
 
-export default class ManifestJSONParser {
+
+export default class ManifestJSONParser extends JSONParser {
 
   constructor(jsonString, collector, {
-    rJSON=RJSON, selfHosted=getConfig().argv.selfHosted,
+    filename=MANIFEST_JSON, rJSON=RJSON, selfHosted=getConfig().argv.selfHosted,
```
Comment (muffinresearch) [inline src/parsers/manifestjson.js]: I wonder if this should be `relaxedJSON` instead of `rJSON` to make it clearer what it represents.

**Comment by muffinresearch** [comment]
r+wc

File: src/parsers/manifestjson.js (hunk 1 above)
Comment (tofumatt) [inline src/parsers/manifestjson.js]: Nope. Or not any more; it was a bit convoluted, where `ManifestJSONScanner.isValid` would be set but then also set using this... I just tidied it up.

File: src/parsers/json.js (hunk 2 above)
Comment (tofumatt) [inline src/parsers/json.js]: Hah, thanks. I improved it but forgot to fix the comment 😄

File: tests/parsers/test.json.js (hunk 3 above)
Comment (tofumatt) [inline tests/parsers/test.json.js]: That's a good question... that should be a separate issue, and I think it's fine, but we could investigate further.

**Pull request merged**